### PR TITLE
TOU: Enable middle auth for MHV

### PR DESCRIPTION
**Pull request by asg5704**
**Note**: Delete the description statements, complete each step. **None are optional**, but can be justified as to why they cannot be completed as written. Provide known gaps to testing that may raise the risk of merging to production.

## Summary
This PR includes the body of work that enables "middle auth" on the Terms of Use page. Specifically it creates the following changes:
- Allows the `/keepalive` to occur on the Terms of Use page, so a user can be signed into the backend to make authenticated calls to the API
- Attempts to save the `redirect_url` query parameter from MHV and save it in session storage to be used later by the Accept/Decline on TOU page (the second time)

## Related issue(s)

- department-of-veterans-affairs/va.gov-team#63933

## Testing done
Unit tests are written however this needs to be tested in staging & below environments

## Screenshots
n/a

## What areas of the site does it impact?
This will impact the Terms of Use for staging & below

### Quality Assurance & Testing

- [ ] I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [ ] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [ ] Linting warnings have been addressed
- [ ] Documentation has been updated ([link to documentation](#) \*if necessary)
- [ ] Screenshot of the developed feature is added
- [ ] [Accessibility testing](https://depo-platform-documentation.scrollhelp.site/developer-docs/wcag-2-1-success-criteria-and-foundational-testing) has been performed

### Error Handling

- [ ] Browser console contains no warnings or errors.
- [ ] Events are being sent to the appropriate logging solution
- [ ] Feature/bug has a monitor built into Datadog or Grafana (if applicable)

### Authentication

- [ ] Did you login to a local build and verify all authenticated routes work as expected with a test user

### :warning: Team Sites (only applies to modifications made to the VA.gov header) :warning:

- [ ] The vets-website header does not contain any web-components
- [ ] I used the [proxy-rewrite steps](https://github.com/department-of-veterans-affairs/vets-website/tree/main/src/applications/proxy-rewrite#local-dev) to test the injected header scenario
- [ ] I reached out in the `#sitewide-public-websites` Slack channel for questions

## Requested Feedback

(OPTIONAL) _What should the reviewers know in addition to the above. Is there anything specific you wish the reviewer to assist with. Do you have any concerns with this PR, why?_

### CODE REVIEW

**Review by joeniquette — APPROVED** [review]
We cant test this, but the code doesnt get deployed to prod for ToU so we are safe from it impacting ToU. The main risk is going to be around autosso and redirect urls for USiP. We should merge this after prod freeze (all items the PR author knows, but documenting for others). 

Alex, just wanted to confirm the questions I have, if you have no concerns with my questions then we are good to merge.

File: src/applications/terms-of-use/containers/TermsOfUse.jsx
Comment (asg5704) [reply src/applications/terms-of-use/containers/TermsOfUse.jsx:30]: Yes we are verifying the parsed URLs https://github.com/department-of-veterans-affairs/vets-website/blob/main/src/applications/terms-of-use/helpers.js

File: src/platform/site-wide/user-nav/containers/AutoSSO.jsx
Comment (asg5704) [reply src/platform/site-wide/user-nav/containers/AutoSSO.jsx:31]: Yes, so we originally had blocked auto sso on the terms of use page because we didn't want the user to automatically be signed in without grabbing where they were redirecting to.

Also, I think Trevor knows how to test it somehow

**Comment by jilladams** [comment]
Noting for @randimays @chriskim2311 PW got tagged here bc of the codeowner rule for `src/platform/site-wide`. However: vsp-identity has a more specific ownership on AutoSSO file in that dir ([here](https://github.com/department-of-veterans-affairs/vets-website/blob/main/.github/CODEOWNERS#L40)) so our review on this is nominal.

**Review by rmessina1010 — APPROVED** [review]
LGTM

**Pull request merged**
